library/pdf_creator.class.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * PdfCreator class
+ *
+ * @package OpenEMR
+ * @link    http://www.open-emr.org
+ * @author  Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2017 Jerry Padgett <sjpadgett@gmail.com>
+ * @license https://www.gnu.org/licenses/agpl-3.0.en.html GNU Affero General Public License 3
+ */
+use Knp\Snappy\Pdf;
+use Symfony\Component\Debug\ExceptionHandler;
+
+class PdfCreator
+{
+
+    private $binaryPath;
+
+    private $tempPath;
+
+    private function getBinaryPath()
+    {
+        global $webserver_root;
+        // This will not necessarily reflect actual machine bus width but php bus size.
+        $intsize = strlen(decbin(~ 0));
+        if (empty(strstr(php_uname("m"), '64'))) {
+            $bit = "32bit";
+        } else {
+            $bit = "64bit";
+        }
+        try {
```

Comment:
Problem with using composer here is I can't find any trusted 1,2 or 3 places that have the binaries. h4cc has linux but is not current. We need somewhere to maintain our dependencies where we have control over them so, this is where I was hoping sub modules could help.
Take nodejs as example. You get what the author thinks you should have and you usually end up with a bunch of bloat. There is more to it than my poor explanation here but I think you get the gist.
I think we should create our own repositories for custom dependencies and composer could be used to install them.